### ISSUE TITLE
GameGuru MAX - Bug - Undo Does Not Restore Move

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#2712]
If you move an object then hit CTRL+Z to undo that move, it does work. In fact, if you add an object, move the object, then press CTRL+Z, undo will remove the added object. Therefore, UNDO does not consider moving an object as a state to undo ... at least in most cases. This make it a pain if you accidently move something and want to just return it to it's previous position, etc.

**Comment by ZakJudges** [comment]
@LeeBamberTGC I believe this occurs when the widget is enabled. Since being updated, you cannot move objects without the widget (#2711)
Do you know if this is by design or not? I can add undo/redo to the new widget code if it is :)

**Comment by ArgentArts** [comment]
Frankly, from a modeler's perspective (and I can only speak for me), if I am using the widget, then the widget is what I am going to use to move and nothing else (as long as he widget is active). We have the "square" areas between axis to move along more than one axis at a time, so, to me, there's no real reason to use 'smart move position' when using the widget. Again, just my opinion.

**Comment by Granada1** [comment]
I live the smart move for placing lights and initial placing of other objects ,a easy to click between options would be cool

**Comment by LeeBamberTGC** [comment]
@ZakJudges If you feel it is an easy add, bounce the idea of RIck but if we are not using those keys I think F1 widget toggle and F2, F3, F4 for position, rotation and scale when you are in widget mode and F2, F3 and F4 for smart, horiz and vert when not in widget mode gives all users instant shortcuts we currently do not have to much-used features!  Rick may want the idea sketched out more before agreeing :)

**Comment by ZakJudges** [comment]
Rick has given the go-ahead. I'll start work on it now :)

**Comment by Granada1** [comment]
Great addition

**Comment by ZakJudges** [comment]
Hi @ArgentArts, there should be a new experimental build up later today that contains a fix for undo, as well as pressing F1 to toggle the widget on and off. Could you let me know if the issue is fixed for you when you update? :)

**Comment by ArgentArts** [comment]
@ZakJudges It seems to work great! I really like it!

The only thing I noticed is if you, for some odd reason, keep the F1 keep depressed, it will continually toggle between modes, flashing the widget on and off. So, if your press on the F1 is a little long, you may accidently go from widget on to widget off to widget back on again and not mean to. I would think that one press, no matter how long, would only turn the widget from on to off or off to on and not constantly toggle it as long as it's held. 

In other words, I think it should require the end-user to have to actually lift their finger and press again to manifest another toggle, if that makes sense.

**Comment by ZakJudges** [comment]
@ArgentArts Thanks for reporting that, I have added a fix so that no matter how long you hold F1, it will only switch modes once. Not sure when the next experimental build will be up, but it will be included in that :)

**Comment by ArgentArts** [comment]
Awesome! Thanks! You rock!
